lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -738,17 +739,17 @@ fromLedgerExUnits
     -> W.ExecutionUnits
 fromLedgerExUnits (Alonzo.ExUnits mem steps) =
     W.ExecutionUnits
-    { executionSteps = steps
-    , executionMemory = mem
+    { executionSteps = fromIntegral steps
+    , executionMemory = fromIntegral mem
```

Comment:
This looks like an unsafe `Natural` to `Word64` conversion.

I believe this is something we _really_ don't want to overflow.

Can we change `W.ExecutionUnits` to also use `Natural` instead of `Word64`?

Reply:
Yes, definitely.